### ISSUE TITLE
Is there any solution to backup and restore data, replace glusterfs nodes ?

### ISSUE THREAD

**Issue by davistran86** [gluster/gluster-kubernetes#405]
Hi, I'm running glusterfs and heketi on openshift cluster, I would like to need some help about how to backup and restore if the glusterfs nodes die ? 
Can I replace a new node with the old one by editing the topology.json and rerun gk-deploy ?

**Comment by jarrpa** [comment]
As far as backup goes. all GlusterFS volumes created by heketi are by default 3-way replica. This means that for any given volume the data is replicated on three different nodes. If you've specified any zones, then they will be as evenly distributed across as many nodes as possible (up to 3 in the replica-3 case).

On the node replacement, I'll quote @obnoxxx:

> You can't specifically tell it to use a concret host for bricks,
> but you can do
> 
> heketi-cli node remove $ID
> 
> and this will do corresponding device remove operations on all
> devices on that node, which will replace the bricks of affected
> volumes by bricks on devices on other nodes. If you have set
> up the zones correctly, or if this is a three-node cluster and you
> use replica-3, then this will automagically use the desired node.

 Let us know if this helps. :)

**Comment by davistran86** [comment]
Thanks @jarrpa , so if 1 node failed, can I use gk-deploy to add another node to existing glusterfs cluster ? like edit topology.json, replace info of failed node inside with new one, and run ./gk-deploy -y ? 

Sorry I'm new to this so still trying to learn how to use it. Thanks a lot :)

**Comment by jarrpa** [comment]
Not quite. You want to use the `heketi-cli` tool. You can find it by running `kubectl exec -it <HEKETI_POD> bash` then using `heketi-cli -s http://localhost:8080 --user admin` to run `heketi-cli` commands. You'll be most likely interested in the `node remove` and `node add` commands. Be sure to read the help messages thoroughly. :)

**Comment by davistran86** [comment]
@jarrpa thanks, got it :)
